rotkehlchen/inquirer.py

```diff
@@ -500,6 +509,102 @@ def find_uniswap_v2_lp_price(
         share_value = numerator / total_supply
         return Price(share_value)
 
+    def find_curve_lp_price(
+        self,
+        lp_token: EthereumToken,
+    ) -> Optional[Price]:
+        """
+        1. Obtain the pool for this token
+        2. Obtain prices for assets in pool
+        3. Obtain the virtual price for share and the balances of each
+        token in the pool
+        4. Calc the price for a share
+        """
+        assert self._ethereum is not None, 'Inquirer ethereum manager should have been initialized'  # noqa: E501
+
+        pools = get_curve_pools()
+        if lp_token.ethereum_address not in pools:
+            return None
+        pool = pools[lp_token.ethereum_address]
+        tokens = []
+        # Translate addresses to tokens
+        try:
+            for asset in pool.assets:
+                # Eth address is translated to weth
+                if asset == '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE':
+                    tokens.append(A_WETH)
+                else:
+                    tokens.append(EthereumToken(asset))
+        except UnknownAsset:
+            return None
+
+        # Get price for each token in the pool
+        prices = []
+        for token in tokens:
+            price = self.find_usd_price(token)
+
+            if price == Price(ZERO):
+                return None
+            prices.append(price)
+
+        # Query virtual price of LP share and balances in the pool for each token
+        contract = EthereumContract(
+            address=pool.pool_address,
+            abi=CURVE_POOL_ABI,
+            deployed_block=0,
+        )
+        call = [(pool.pool_address, contract.encode(method_name='get_virtual_price'))]
+        call += [
+            (pool.pool_address, contract.encode(method_name='balances', arguments=[i]))
+            for i in range(len(pool.assets))
+        ]
+
+        output = multicall_2(
+            ethereum=self._ethereum,
+            require_success=True,
+            calls=call,
+        )
+        # Deserialice information obtained in the multicall execution
+        data = []
+        data.append(contract.decode(output[0][1], 'get_virtual_price')[0])
```

Comment:
what if output does not have this structure? It would raise either `IndexError` or `TypeError`